backend/api.test/EventHandlers/TestMissionScheduler.cs

```diff
@@ -103,20 +100,30 @@ public void Dispose()
         /// </summary>
         /// <param name="preStatus"></param>
         /// <param name="postStatus"></param>
+        /// <param name="mission"></param>
         private async void AssertExpectedStatusChange(
             MissionStatus preStatus,
-            MissionStatus postStatus
+            MissionStatus postStatus,
+            Mission mission
         )
         {
             // ARRANGE
 
             var cts = new CancellationTokenSource();
 
             // Add Scheduled mission
-            await _missionService.Create(ScheduledMission);
+            await _missionService.Create(mission);
+
+            _robotControllerMock.RobotServiceMock
+                .Setup(service => service.ReadById(mission.Robot.Id))
+                .Returns(async () => mission.Robot);
```

Comment:
Alot of methods missing await, might wanna ignore warning or add await

Reply:
Fixed